README.md

```diff
@@ -276,6 +276,25 @@ Using this method you can run as many shell commands as you need with whatever
 logic you need. Though it is suggested that if it gets too long you might want
 to wrap it in a shell script, deploy and run that.
 
+#### Shell Commands and Exec Mode
+
+Using the system shell based command has one additional caveat when used for
+the Exec mode process (the managed, executed process to which it will propagate
+signals). That is to get signals to work correctly means not only does anything
+the shell runs need to handle signals, but the shell itself needs to. This
+needs to managed by you as shells will exit upon receiving most signals.
+
+A common example of this would be wanting the SIGHUP signal to trigger a reload
+of the underlying process and to be ignored by the shell process. To get this
+you need to trap the signal in the shell with no action. For example if you had
+an underlying nginx process you wanted to run with a shell command and have the
+shell ignore it you'd do..
+
+`command = "trap '' HUP; /usr/sbin/nginx -c /etc/nginx/nginx.conf"`
+
+The `trap '' HUP;` bit is enough to get the shell to ignore the HUP signal. See
+your shell's documentation on `trap` for more details on this.
+
```

Comment:
This example really helped me here! Just to make sure I'm understanding correctly, the idea is that if you didn't add the `trap '' HUP;`, a SIGHUP signal would have likely stopped the shell process running nginx when what we want is just nginx to process the SIGHUP.

Reply:
Maybe if having the non-working version for as a comparison would help? Like saying:

`command = "/usr/sbin/nginx -c /etc/nginx/nginx.conf"`
When receiving a SIGHUP, the shell process would stop, which also stops the nginx process.

`command = "trap '' HUP; /usr/sbin/nginx -c /etc/nginx/nginx.conf"`
When receiving a SIGHUP, the shell will ignore the signal and instead nginx will process the signal and reload.

(I'm not confident on my terminology here!)

Reply:
Thought of another way I wanted to document as a way of dealing with this.. using `exec` to have the sub-process replace the shell process. I'm going to add this as well as add a bit more on what happens if you don't do it.

Reply:
Looked over your latest changes and they make sense to me!